sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_client_manager.py

```diff
@@ -303,37 +309,75 @@ def __init__(
+        self._active_clients: List[_ConfigurationClientWrapper] = []
 
         if connection_string and endpoint:
             self._replica_clients.append(
                 _ConfigurationClientWrapper.from_connection_string(
                     endpoint, connection_string, user_agent, retry_total, retry_backoff_max, **self._args
                 )
             )
-            self._setup_failover_endpoints()
-            return
-        if endpoint and credential:
+        elif endpoint and credential:
             self._replica_clients.append(
                 _ConfigurationClientWrapper.from_credential(
                     endpoint, credential, user_agent, retry_total, retry_backoff_max, **self._args
                 )
             )
-            self._setup_failover_endpoints()
+        else:
+            raise ValueError("Please pass either endpoint and credential, or a connection string with a value.")
+
+    def get_next_client(self) -> Optional[_ConfigurationClientWrapper]:
+        """
+        Get the next client to be used for the request. find_active_clients needs be called before this method or None
+        client will be returned. If load_balancing_enabled isn't enabled the first active client will be returned,
+        which is the provided endpoint unless it has failed. If load_balancing_enabled is enabled the next client in
+        the list will be returned.
+
+        :return: The next client to be used for the request.
+        """
+        if not self._active_clients:
+            self._last_active_client_name = ""
+            return None
+        if not self._load_balancing_enabled:
+            for client in self._active_clients:
+                if client.is_active():
+                    return client
+            return None
+        next_client = self._active_clients.pop(0)
+        self._last_active_client_name = next_client.endpoint
+        return next_client
+
+    def find_active_clients(self):
+        """
+        Figures out the current active clients, if _load_balancing_enabled is enabled the start of the list will be the
+        one after the last active client used.
+        """
+        active_clients = [client for client in self._replica_clients if client.is_active()]
+
+        self._active_clients = active_clients
+        if not self._load_balancing_enabled or len(self._last_active_client_name) == 0:
             return
-        raise ValueError("Please pass either endpoint and credential, or a connection string with a value.")
+        for i, client in enumerate(active_clients):
+            if client.endpoint == self._last_active_client_name:
+                swap_point = (i + 1) % len(active_clients)
+                self._active_clients = active_clients[swap_point:] + active_clients[:swap_point]
+                return
+
+    def get_client_count(self) -> int:
+        return len(self._replica_clients)
 
     def refresh_clients(self):
```

Comment:
We are not calling this method anywhere now. As a result, we are not calling `find_auto_failover_endpoints` anywhere in sync code

Reply:
Added it back, accidentally removed it.